pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -3378,7 +3402,27 @@ public void updateArray(String columnName, Array x) throws SQLException {
         return type.cast(getLocalDateTime(columnIndex));
       } else {
         throw new PSQLException(GT.tr("conversion to {0} from {1} not supported", type, sqlType),
-                PSQLState.INVALID_PARAMETER_VALUE);
+            PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == ZonedDateTime.class) {
+      if (sqlType == Types.TIMESTAMP_WITH_TIMEZONE || sqlType == Types.TIMESTAMP) {
+        Timestamp timestampValue = getTimestamp(columnIndex);
+        if (wasNull()) {
+          return null;
+        }
+        long time = timestampValue.getTime();
+        if (time == PGStatement.DATE_POSITIVE_INFINITY) {
+          return type.cast(ZonedDateTime.of(LocalDateTime.MAX, ZoneId.of("UTC")));
```

Comment:
You should be able to use `ZoneOffset.UTC` here an in the other places.

Reply:
@marschall thx! fixed
@vlsi so what do we do? People will just complain that we don't have the method implemented.

Reply:
@davecramer the specification only requires java.time.LocalDate, java.time.LocalTime, java.time.LocalDateTime, java.time.OffsetTime and java.time.OffsetDatetime...

Reply:
H2 and connector-j only support `OffsetDateTime`, AFAIK mssql-jdbc and hsqldb also only support `OffsetDateTime`.
ojdbc supports `ZonedDateTime` but there the storage is different and Oracle can return the time zone not just the offset.
I haven't looked further but I expect the majority of JDBC drivers do not support `ZonedDateTime`.

That doesn't mean you shouldn't but it also means that people can not generally expect a JDBC driver to support `ZonedDateTime` since most don't.